csharp/device/Microsoft.Azure.Devices.Client/HttpDeviceClient.cs

```diff
@@ -262,13 +461,30 @@ protected override Task OnCompleteAsync(string lockToken)
             var eTag = new ETagHolder { ETag = lockToken };
 
             return this.httpClientHelper.DeleteAsync(
-                GetRequestUri(this.deviceId, CommonConstants.DeviceBoundPathTemplate + "/{0}".FormatInvariant(lockToken), null),
+                GetRequestUri(this.deviceId, CommonConstants.DeviceBoundPathTemplate + "/{0}".FormatInvariant(lockToken), new Dictionary<string, string> { { "reject", null } }),
```

Comment:
Nothing to do with NETMF. Probably a bad merge...
I've check the code in the develop branch and commit the correct line.